modular_skyrat/modules/modular_items/lewd_items/code/lewd_items/shibari_stand.dm

```diff
@@ -0,0 +1,269 @@
+				return FALSE
+		else
+			if(!do_after(user, 10 SECONDS, buckledmeme))
+				return FALSE
+
+		if(!is_user_buckle_possible(buckledmeme, user, check_loc))
+			return FALSE
+
+		if(!(istype(hooman.w_uniform, /obj/item/clothing/under/shibari_fullbody)))
+			to_chat(user, span_warning("You'll need to completely tie their body!"))
+			return FALSE
+		if(!istype(user.get_active_held_item(), /obj/item/stack/shibari_rope))
+			to_chat(user, span_warning("You'll need to be holding shibari ropes to tie them to the stand!"))
+			return FALSE
+
+		if(buckle_mob(buckledmeme, check_loc = check_loc))
+			var/obj/item/stack/shibari_rope/rope = user.get_active_held_item()
+			ropee = new()
+			ropee.current_color = rope.current_color
+			ropee.update_icon_state()
+			ropee.update_icon()
+			rope.use(1)
+			add_overlay(shibari_shadow_overlay)
+			add_rope_overlays(ropee.current_color, hooman?.dna?.species?.mutant_bodyparts["taur"])
+			buckledmeme.visible_message(span_warning("[user] tied [buckledmeme] to [src]!"),\
+				span_userdanger("[user] tied you to [src]!"),\
+				span_hear("You hear ropes being completely tightened."))
+	else
+		to_chat(user, span_warning("You cannot buckle yourself to this stand, there is no way that level of self-bondage exists!"))
+		return FALSE
+
+/obj/structure/chair/shibari_stand/deconstruct()
+	qdel(src)
+	return TRUE
+
+/obj/structure/chair/shibari_stand/proc/add_rope_overlays(color, taur)
+	cut_overlay(shibari_rope_overlay)
+	cut_overlay(shibari_rope_overlay_behind)
+	if(taur)
+		shibari_rope_overlay = mutable_appearance('modular_skyrat/modules/modular_items/lewd_items/icons/obj/lewd_structures/shibari_stand.dmi', "ropes_above_[color]_snek", ABOVE_MOB_LAYER)
+	else
+		shibari_rope_overlay = mutable_appearance('modular_skyrat/modules/modular_items/lewd_items/icons/obj/lewd_structures/shibari_stand.dmi', "ropes_above_[color]", ABOVE_MOB_LAYER)
+	shibari_rope_overlay_behind = mutable_appearance('modular_skyrat/modules/modular_items/lewd_items/icons/obj/lewd_structures/shibari_stand.dmi', "ropes_behind_[color]", BELOW_MOB_LAYER)
+	add_overlay(shibari_rope_overlay)
+	add_overlay(shibari_rope_overlay_behind)
+
+/obj/structure/chair/shibari_stand/post_buckle_mob(mob/living/buckledmeme)
+	buckledmeme.pixel_y = buckledmeme.base_pixel_y + 4
+	buckledmeme.pixel_x = buckledmeme.base_pixel_x
+	buckledmeme.layer = BELOW_MOB_LAYER
+
+	if(LAZYLEN(buckled_mobs))
+		if(ishuman(buckled_mobs[1]))
+			current_mob = buckled_mobs[1]
+
+	if(current_mob)
+		if(current_mob.handcuffed)
+			current_mob.handcuffed.forceMove(loc)
+			current_mob.handcuffed.dropped(current_mob)
+			current_mob.set_handcuffed(null)
+			current_mob.update_handcuffed()
+		current_mob.set_handcuffed(new /obj/item/restraints/handcuffs/milker/shibari(current_mob))
+		current_mob.handcuffed.parented_struct = src
+		if(HAS_TRAIT(current_mob, TRAIT_ROPEBUNNY))
+			current_mob.handcuffed.breakouttime = 4 MINUTES
+		current_mob.update_abstract_handcuffed()
```

Comment:
I just feel like this should call the parent proc.

Reply:
doesn't want it to inherit stuff from the chair